test/ibmq/test_filter_backends.py

```diff
@@ -56,15 +56,27 @@ def test_filter_least_busy(self, provider):
         filtered_backends = least_busy(backends)
         self.assertTrue(filtered_backends)
 
+    # TODO: Refactor this test to be more accurate. Not completely sure if this covers it.
+    @requires_provider
+    def test_get_properties_filter_none(self, provider):
+        """Test retrieving properties from backends without filters."""
+        backends = provider.backends(simulator=False)
+
+        for backend in backends:
+            with self.subTest(backend=backend):
+                properties = backend.properties()
+                self.assertTrue(properties.last_update_date.date() == datetime.now().date())
+
     @requires_provider
     def test_get_properties_filter_date(self, provider):
-        """Test retrieving properties from a backend filtered by date."""
+        """Test retrieving properties from backends filtered by date."""
         backends = provider.backends(simulator=False)
 
         my_filter = {'last_update_date': {'lt': '2019-01-01T00:00:00.000'}}
         filter_date = datetime.fromisoformat(my_filter['last_update_date']['lt'])
 
         for backend in backends:
             with self.subTest(backend=backend):
-                properties = backend.properties()
-                self.assertTrue(properties.last_update_date < filter_date)
+                properties = backend.properties(db_filter=my_filter)
```

Comment:
This is the line that fails. When the iteration reaches backend='ibmqx2', the attempt to get the backend properties throws the following Exception:

SubTest error: Traceback (most recent call last): File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/case.py", line 59, in testPartExecutor yield File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/case.py", line 533, in subTest yield File "/Users/luisgarcia/IBMQ/qiskit-ibmq-provider/test/ibmq/test_filter_backends.py", line 80, in test_get_properties_filter_date properties = backend.properties(db_filter=my_filter) File "/Users/luisgarcia/IBMQ/qiskit-ibmq-provider/qiskit/providers/ibmq/ibmqbackend.py", line 96, in properties self._properties = BackendProperties.from_dict(api_properties) File "/Users/luisgarcia/IBMQ/venvs/venv-qiskit-ibmq-provider/lib/python3.7/site-packages/qiskit/validation/base.py", line 352, in from_dict ex.messages, ex.field_names, ex.fields, ex.data, **ex.kwargs) from None qiskit.validation.exceptions.ModelValidationError: {'qubits': ['Missing data for required field.'], 'general': {0: {'date': ['Missing data for required field.'], 'unit': ['Missing data for required field.'], 'name': ['Missing data for required field.'], 'value': ['Missing data for required field.']}}, 'last_update_date': ['Missing data for required field.'], 'backend_name': ['Missing data for required field.'], 'gates': {0: {'qubits': ['Missing data for required field.'], 'parameters': {0: {'date': ['Missing data for required field.'], 'unit': ['Missing data for required field.'], 'name': ['Missing data for required field.'], 'value': ['Missing data for required field.']}}, 'gate': ['Missing data for required field.']}}, 'backend_version': ['Missing data for required field.']}

Reply:
This is likely due to no calibrations being available for the combination of timestamp + device used in the test - which actually is something that might happen "often enough" that we can intercept it during `backend.properties()`, in order to avoid the cryptic message reaching the user.

Reply:
Yeah, that seems to be the issue. After using a more recent date, there are no tests that fail. So, I will look into intercepting these cases.